packages/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,502 @@
+<template>
+  <KPop
+    :hide-popover="hidePopover"
+    class="k-datetime-picker"
+    placement="bottomStart"
+    width="auto"
+    hide-caret
+    position-fixed
+    @closed="handleClose"
+  >
+    <KButton
+      :is-rounded="false"
+      size="large"
+      class="timepicker-input"
+    >
+      {{ abbreviatedDisplay }}
+    </KButton>
+    <template
+      v-if="!hidePopover"
+      #content>
+      <KSegmentedControl
+        v-if="hasRelativeTimeframes && hasCalendar"
+        v-model="tabName"
+        :options="[
+          { label: 'Relative', value: 'relative' },
+          { label: 'Custom', value: 'custom' }
+        ]"
+        class="w-100 mb-5"
+        data-testid="analytics-time-toggle"
+      />
+      <p v-if="!showCalendar">{{ fullRangeDisplay }}</p>
+      <DatePicker
+        v-if="hasCalendar && showCalendar"
+        v-model="selectedCalendarRange"
+        :is-range="isRange"
+        :select-attribute="calendarSelectAttributes"
+        :model-config="modelConfig"
+        :mode="calendarMode"
+        is-expanded
+      />
+      <div
+        v-else-if="hasRelativeTimeframes && hasTimePeriods"
+        class="d-flex flex-column"
+      >
+        <div
+          v-for="(item, index) in timePeriods"
+          :key="`section-${index}`"
+          class="timeframe-section d-flex flex-column"
+        >
+          <h4>{{ item.section }}</h4>
```

Comment:
question: does this need a `v-if` in case the user doesn't provide a string for the `item.section` to prevent the empty tag from rendering?
```suggestion
          <h4 v-if="item.section">{{ item.section }}</h4>
```